### PR TITLE
FLA-751 added links to files in the DocumentList component

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

https://justice.gov.bc.ca/jira/browse/FLA-751
- Added hyperlinks on files listed in the DocumentList component
- Added a new service call to the backend API
- Added unit tests

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

yarn lint
yarn test

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by alexjoybc — COMMENTED** [review]
that looks really good @KDNeufeld

File: src/frontend/efiling-frontend/src/domain/package/package-review/DocumentList.js [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
 import PropTypes from "prop-types";
 
+import { errorRedirect } from "../../../modules/helpers/errorRedirect";
+import { getSubmittedDocument } from "./PackageReviewService";
 import "./DocumentList.css";
 
-export default function DocumentList({ documents }) {
+export default function DocumentList({ packageId, documents }) {
+  function handleClickFile(document) {
+    getSubmittedDocument(packageId, document).catch((err) => {
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/domain/package/package-review/DocumentList.js:10]: That looks really good, is it the service suppose to save the file locally or should it return the blob, don't want to change it now, but something to thing about. or the name of the function should be `saveSubmittedDocumentLocal` or something like that, what do you think ?

File: src/frontend/efiling-frontend/src/domain/package/package-review/DocumentList.js [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+      errorRedirect(sessionStorage.getItem("errorUrl"), err);
+    });
+  }
+
+  function handleKeyDownFile(e, document) {
+    if (e && e.keyCode === 13) {
+      getSubmittedDocument(packageId, document).catch((err) => {
+        errorRedirect(sessionStorage.getItem("errorUrl"), err);
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/domain/package/package-review/DocumentList.js:18]: Duplicate of line 11, not a big deal...

File: src/frontend/efiling-frontend/src/domain/package/package-review/PackageReview.js [hunk 3]
```diff
@@ -114,10 +114,12 @@ export default function PackageReview({ page: { header, packageId } }) {
     });
   }
 
-  function handleKeyDown() {
-    getSubmissionSheet(packageId).catch((err) => {
-      errorRedirect(sessionStorage.getItem("errorUrl"), err);
-    });
+  function handleKeyDown(e) {
+    if (e && e.keyCode === 13) {
+      getSubmissionSheet(packageId).catch((err) => {
```
Comment (alexjoybc) [inline src/frontend/efiling-frontend/src/domain/package/package-review/PackageReview.js:119]: same here for the name of the method `saveSubmissonSheetLocal` ?

**Review by alexjoybc — CHANGES_REQUESTED** [review]
Awesome contribution @KDNeufeld , just a couple of questions

File: src/frontend/efiling-frontend/src/domain/package/package-review/DocumentList.js (hunk 1 above)
Comment (KDNeufeld) [reply src/frontend/efiling-frontend/src/domain/package/package-review/DocumentList.js]: Good suggestion.  Refactored both this method and getSubmissionSheet to downloadSubmittedDocument and downloadSubmissionSheet respectively.

File: src/frontend/efiling-frontend/src/domain/package/package-review/PackageReview.js (hunk 3 above)
Comment (KDNeufeld) [reply src/frontend/efiling-frontend/src/domain/package/package-review/PackageReview.js]: Done. Refactored.

File: src/frontend/efiling-frontend/src/domain/package/package-review/DocumentList.js (hunk 2 above)
Comment (KDNeufeld) [reply src/frontend/efiling-frontend/src/domain/package/package-review/DocumentList.js:18]: Yeah, we need to come up with a better way for error handling globally.  Just redirecting to the error view is not always appropriate.  We instead should show a toast error banner, or have a reserved notification area, or ...

Comment (alexjoybc) [reply src/frontend/efiling-frontend/src/domain/package/package-review/DocumentList.js:18]: Yes let's add one for it

**Review by alexjoybc — APPROVED** [review]
Looks good thanks @KDNeufeld

**Pull request merged**